### ISSUE TITLE
Submit Event not getting triggered for form

### ISSUE THREAD

**Issue by ShadabFaiz** [testing-library/react-testing-library#755]
- `@testing-library/react` version: 9.3.2
- Testing Framework and version: Jest v 24.9.0
  
- DOM Environment: jest-dom v4.2.4
  

### Relevant code or config:

```js
it('submits with button outside of form', () => {
  const onSubmit = () => {
    console.log('onSubmit')
  }
  const { getByText } = render(
    <div>
      <form id="form-id" onSubmit={onSubmit}>
        <input />
      </form>
      <button form="form-id" type="submit">
        Submit
      </button>
    </div>
  )
  fireEvent.submit(getByText('Submit'))
})
```

### What you did:
Ran the test on the above code.

### What happened:
Submit event is not fired.

### Reproduction:
Code mentioned above.

### Problem description:
When the submit button is outside the form, then fireEvent.click does not trigger the submit event on form. During the normal application running, submit button does submit event. 

### Suggested solution:
FireEvent.click should trigger submit event for submit buttons that are outside the form.

**Comment by marcosvega91** [comment]
Hi @ShadabFaiz thanks for opening the issue :).

I think that `submit` event should be fired on form but not on `button`. You should instead `fire click` on the `button`, the click will fire `submit` on the `form` even `jsdom` is not supporting submit https://github.com/jsdom/jsdom/issues/1937


https://codesandbox.io/s/fervent-banach-kzp9r?file=/src/__tests__/index.js

**Comment by Strider9494** [comment]
@marcosvega91 it doesn't work with click event too 

```
 it('submits with button outside of form', async () => {
    const mockedFn = jest.fn(() => {
      console.log('onSubmit');
    });

    const { getByText } = render(
      <div>
        <form id="form-id" onSubmit={mockedFn}>
          <input />
        </form>
        <button form="form-id" type="submit">
          Submit
        </button>
      </div>
    );
    fireEvent.click(getByText('Submit'));
    await wait(() => {
      expect(mockedFn).toHaveBeenCalledTimes(1);
    });
  });
```

**Comment by marcosvega91** [comment]
Hi @Strider9494 I have updated the codesandbox with your code and is still working 🤔

**Comment by canastro** [comment]
@marcosvega91 Try to move the button within the form

**Comment by marcosvega91** [comment]
Hi @canastro I have moved the button inside the form with no issue. 

https://codesandbox.io/s/fervent-banach-kzp9r?file=/src/__tests__/index.js

**Comment by canastro** [comment]
Sorry, I meant to tag @Strider9494.

**Comment by stilllife00** [comment]
Just an info that might help, I had a similar issue using formik and got it working by adding:
```
@jest-environment jest-environment-jsdom-sixteen
```
at the beginning of the file. I guess I have some conflicting jest/jsdom versions

**Comment by argipap** [comment]
Hi guys,

I have a similar issue in testing Login form with Formik.
After changing input text for email and password fields and then click the submit button, toHaveBeenCalled() method in jest.fn function returns that received zero number of calls.
Reproduced in the codesandbox:
https://codesandbox.io/s/loginformtesting-br1tp?file=/src/__tests__/LoginForm.test.js

Any help would be really appreciated.

Thx,
Argiris

**Comment by eps1lon** [comment]
This particular issue is likely caused by using older `jsdom` environments (via `jest`). The bug was fixed in https://github.com/jsdom/jsdom/pull/2369 which was released in `jsdom@12.2.0`. However, `jest` 24  uses `jsdom` 11 so you either need to upgrade your `jest` version or use the workaround from @stilllife00  described in https://github.com/testing-library/react-testing-library/issues/755#issuecomment-731205845.

Verified described behavior works in Chrome 87: https://codesandbox.io/s/objective-lake-yr8yx?file=/src/index.js
Verified described behavior works in `jsdom` 16: https://runkit.com/eps1lon/5fce1673c6e29d001ab38686

**Comment by machernator** [comment]
Although this is already closed, the simplest solution to trigger submit from outside a form is to add the form attribute to a button. It should point to the id of the form. Then it belongs to the form, no matter where in the DOM it lives. The same goes for input/select/textarea.
